docs: add os upgrades guide DOC-1533 (#5647)
spectrocloud/librarium: docs/docs-content/deployment-modes/agent-mode/manage-agent/os-upgrades.md

   [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) to execute upgrades on a repeating
   schedule. You can find some common examples of Cron schedules in the following table.

   | **Expression** | **Description**                                        |
   | -------------- | ------------------------------------------------------ |
   | `0 0 1 1 *`    | Once a year at midnight of 1 January                   |
   | `0 0 1 * *`    | Once a month at midnight of the first day of the month |
   | `0 0 * * 0`    | Once a week at midnight on Sunday morning              |
   | `0 0 * * *`    | Once a day at midnight                                 |
   | `0 * * * *`    | Once an hour at the beginning of the hour              |

   Execute the following command in your terminal, replacing the placeholder with your preferred Cron schedule. The
   command saves your chosen schedule to the `SYSTEM_UPGRADE_SCHEDULE` variable.

   ```shell
   export SYSTEM_UPGRADE_SCHEDULE="REPLACE ME"
   ```

8. Execute the following command in your terminal, replacing the placeholder with a node label of your choice. This

>    variable allows you to customise which nodes should be periodically updated. The command saves your label to the

Reviewer: ```suggestion
   variable allows you to customize which nodes should be periodically updated. The command saves your label to the
```

   `SYSTEM_UPGRADE_NODE_LABEL` variable.

   ```shell
   export SYSTEM_UPGRADE_NODE_LABEL="REPLACE ME"
   ```

9. Apply the node label to all the nodes that you want updated. Execute the command by replacing the placeholder with
   the name of the node. Repeat for each node you want to upgrade.

   ```shell
   kubectl label node REPLACE-ME $SYSTEM_UPGRADE_NODE_LABEL=
   ```

   :::info

   Nodes are drained, upgraded and rebooted one by one. Ensure that your cluster has enough resources to perform rolling
   upgrades in order to avoid outages.

   :::

10. Save your upgrade scripts to a file titled `upgrades.sh`. You can provide any instructions that you want to execute
    on system upgrade and reboot. The following example provides upgrade instructions for Ubuntu, but you can modify

>     them to work according to your edge host operating system. The command creates the `upgrades.sh` file in your local

Reviewer: ```suggestion
    them to work according to your host operating system. The command creates the `upgrades.sh` file in your local
```

                                  export KUBECONFIG=/run/kubeconfig
                                  kubectl get plan os-upgrade-plan --namespace $SYSTEM_UPGRADE_NAMESPACE
                                  if [ \$? -eq 0 ]; then
                                    echo "Upgrade plan exists. Retrigger it."
                                    VERSION="os-upgrade-plan-\$(date +%Y%m%d%H%M%S)"
                                    kubectl patch plan os-upgrade-plan --namespace $SYSTEM_UPGRADE_NAMESPACE --type=json --patch="[{\"op\": \"replace\", \"path\": \"/spec/version\", \"value\": \"\${VERSION}\"}]"
                                  else
                                    echo "Upgrade plan does not exist. Create it."
                                    kubectl get secret os-upgrade-plan --namespace $SYSTEM_UPGRADE_NAMESPACE --output go-template='{{ index .data "plan.yaml" | base64decode }}' | kubectl apply --filename -
                                    fi
                        restartPolicy: OnFailure
    EOF
    ```

12. Navigate back to [Palette](https://console.spectrocloud.com) in your browser. Select **Profiles** from the **left
    Main Menu**.

> 13. Select the cluster profile corresponding to your edge cluster.

Reviewer: ```suggestion
13. Select the cluster profile corresponding to your agent mode cluster.
```

14. Click on the version **drop-down Menu**. Select the **Create new version** option. Fill in the **Version** input and
    click **Confirm** to create a new version of your cluster profile. The new profile version opens.

15. Click **Add manifest**. The manifest editor appears. Fill in the **Layer name** input field. Then, click **New
    Manifest**. Input a name for the manifest file. Click on the check or press Enter to open the editor.

16. Paste the contents of the `upgrades.yaml` file that you have created in **Step 11**. Click **Confirm Updates** to
    save your manifest. Then, click **Save Changes** to save your manifest to the cluster profile.

17. Navigate to the **left Main Menu** and select **Clusters**.

18. Select your cluster to access the cluster details page.

19. Click on the **Profiles** tab.

20. Select the newly created version of your cluster profile. Click **Save**.

Palette applies your manifest to the cluster. The Kubernetes resources responsible for the system upgrade are created in
the `system-upgrade-xxx` namespace.

## Validate

1. Log in to [Palette](https://console.spectrocloud.com).

> 2. Navigate to the **left Main Menu** and select **Clusters**.

Reviewer 2: ```suggestion
2. Navigate to the left **Main Menu** and select **Clusters**.
```

3. Select your cluster to access the cluster details page.

4. Click **Settings** and select **Download Logs** from the dropdown. The **Download Logs** window appears.

5. Ensure that **Node Logs** is selected and click **Download**. The logs will take a few minutes to generate. A
   download link will appear once the logs are ready.

> The download contains a zip archive of files. Details of a all executed upgrades are in a folder with the same name as

Reviewer: ```suggestion
The download contains a zip archive of files. Details of all executed upgrades are in a folder with the same name as
```

> your system upgrade namespace. You can search for executions of the `os-upgrade-plan` in this file. You can add further

Reviewer 2: I would be nice to see an example of the logs (in a code snippet, for ex). Are they super lengthy?

Author: Added example.
